### PR TITLE
Add support for disallowed additionalProperties on JSONSchema

### PR DESCRIPTION
**Pull request by kdisneur**

### CODE REVIEW

**Comment by kdisneur** [comment]
Task linked: [CU-f107w4 Don't use json.Decoder for parsing request body](https://app.clickup.com/t/f107w4)

File: platform/web/tests/decode_with_json_schema_test.go [hunk 1]
```diff
@@ -40,58 +40,47 @@ var testCases = []decodeWithJSONSchemaTestData{
 		},
 	},
 	{
-		Name:           "when a parameter has an incorrect datatype",
+		Name:           "when_a_parameter_has_an_incorrect_datatype",
```
Comment (julien-leclercq) [inline platform/web/tests/decode_with_json_schema_test.go:43]: Must the test names be snaked cased now ?

**Review by julien-leclercq — APPROVED** [review]

File: platform/web/tests/decode_with_json_schema_test.go (hunk 1 above)
Comment (kdisneur) [reply platform/web/tests/decode_with_json_schema_test.go:43]: I did it based on this answer: https://github.com/fewlinesco/catalog/pull/443#discussion_r621895160

Comment (julien-leclercq) [reply platform/web/tests/decode_with_json_schema_test.go:43]: yep saw it a bit later :)

Comment (kdisneur) [reply platform/web/tests/decode_with_json_schema_test.go:43]: Based on the answer, I thought the team already settled on what should be done. If not the case, we can bring the topic to Slack or ping the others.

Comment (julien-leclercq) [reply platform/web/tests/decode_with_json_schema_test.go:43]: Itmight have happenned when I was not here

Comment (Kemosabert) [reply platform/web/tests/decode_with_json_schema_test.go:43]: https://fewlines.slack.com/archives/CBQTY8P0T/p1611736007091000

**Review by Kemosabert — APPROVED** [review]

File: platform/web/tests/decode_with_json_schema_test.go (hunk 1 above)
Comment (julien-leclercq) [reply platform/web/tests/decode_with_json_schema_test.go:43]: well it's even a proposition of mine haha

**Comment by tgautier** [comment]
Task linked: [CU-hv2hb7 Update go-pkg to use `json.Unmarshal` instead of `json.NewDecoder`](https://app.clickup.com/t/hv2hb7)

**Pull request merged**